mrmustard/physics/bargmann.py

```diff
@@ -103,3 +104,184 @@ def wigner_to_bargmann_U(X, d):
     N = X.shape[-1] // 2
     A, B, C = wigner_to_bargmann_Choi(X, math.zeros_like(X), d)
     return A[2 * N :, 2 * N :], B[2 * N :], math.sqrt(C)
+
+
+def complex_gaussian_integral(
```

Comment:
I just noticed one critical issue related to this function. 

One example to explain:

dgate1 = Dgate(x=[0.1,0.2], modes=[0,7])
dgate2 = Dgate(x=[0.5,-0.5], modes=[0,88])
circuit = dgate1 >> dgate2
new_component = simulator.run(circuit)

The internal calculation is b vector of dgate1 [0.1, 0.2, -0.1, -0.2]
and the b vector of dgate2 [0.5, -0.5, -0.5, 0.5].

After this complex_integral, I was expected to have the result like [0.6, 0.2, -0.5, -0.6, -0.2, 0.5]. 
This is what I've got : [ 0.2+0.j, -0.6+0.j, -0.2+0.j,  0.6+0.j, -0.5+0.j,  0.5+0.j].

I believe that we need a protocol to reorder them in a "reasonable way".

Reply:
I don't think the ordering issue is related to this PR. The result of an inner product with @ depends on what you contract. After a contraction the right order must be restored, but Bargmann cannot be the one who does it because it doesn't have access to the wires.